wp-includes/mysql/class-mysql-lexer.php

```diff
@@ -0,0 +1,2752 @@
+		$this->n              = $this->input[ $this->position + 1 ] ?? null;
+	}
+
+	public function is_sql_mode_active( int $mode ): bool {
+		return ( $this->sql_modes & $mode ) !== 0;
+	}
+
+	public function get_server_version() {
+		return $this->server_version;
+	}
+
+	public static function get_token_id( string $token_name ): int {
+		if ( 'ε' === $token_name ) {
+			return self::EMPTY_TOKEN;
+		}
+		return constant( self::class . '::' . $token_name );
+	}
+
+	public static function get_token_name( $token_type ): string {
+		$reflection = new ReflectionClass( self::class );
+		$constants  = array_reverse( $reflection->getConstants() ); // some constants can conflict, tokens are at the end
+		$token_name = array_search( $token_type, $constants, true );
+		return $token_name ? $token_name : '<INVALID>';
+	}
+
+	public function get_text(): string {
+		return $this->text;
+	}
+
+	public function get_next_token() {
+		do {
+			$this->next_token();
+		} while ( null === $this->token_instance || self::CHANNEL_HIDDEN === $this->token_instance->channel );
+		return $this->token_instance;
+	}
+
+	private function next_token() {
+		$this->text           = '';
+		$this->type           = null;
+		$this->token_instance = null;
+		$this->channel        = self::CHANNEL_DEFAULT;
+
+		$la  = $this->c;
+		$la2 = $this->n;
+
+		if ( "'" === $la ) {
+			$this->single_quoted_text();
+		} elseif ( '"' === $la ) {
+			$this->double_quoted_text();
+		} elseif ( '`' === $la ) {
+			$this->back_tick_quoted_id();
+		} elseif ( $this->is_digit( $la ) ) {
+			$this->number();
+		} elseif ( '.' === $la ) {
+			if ( $this->is_digit( $la2 ) ) {
+				$this->number();
+			} else {
+				$this->consume();
+				$this->type = self::DOT_SYMBOL;
+			}
+		} elseif ( '=' === $la ) {
+			$this->consume();
+			$this->type = self::EQUAL_OPERATOR;
+		} elseif ( ':' === $la ) {
+			if ( '=' === $la2 ) {
+				$this->consume(); // Consume the ':'.
+				$this->consume(); // Consume the '='.
+				$this->type = self::ASSIGN_OPERATOR;
+			} else {
+				$this->consume();
+				$this->type = self::COLON_SYMBOL;
+			}
+		} elseif ( '<' === $la ) {
+			if ( '=' === $la2 ) {
+				if ( '>' === $this->la( 3 ) ) {
+					$this->consume(); // Consume the '<'.
+					$this->consume(); // Consume the '='.
+					$this->consume(); // Consume the '>'.
+					$this->type = self::NULL_SAFE_EQUAL_OPERATOR;
+				} else {
+					$this->consume(); // Consume the '<'.
+					$this->consume(); // Consume the '='.
+					$this->type = self::LESS_OR_EQUAL_OPERATOR;
+				}
+			} elseif ( '>' === $la2 ) {
+				$this->consume(); // Consume the '<'.
+				$this->consume(); // Consume the '>'.
+				$this->type = self::NOT_EQUAL_OPERATOR;
+			} elseif ( '<' === $la2 ) {
+				$this->consume(); // Consume the '<'.
+				$this->consume(); // Consume the '<'.
+				$this->type = self::SHIFT_LEFT_OPERATOR;
+			} else {
+				$this->consume();
+				$this->type = self::LESS_THAN_OPERATOR;
+			}
```

Comment:
How about moving the `$this->consume();` calls right after the check where we confirmed what's the lookahead token? It seems more readable and shorter.

```suggestion
			$this->consume(); // Consume the '<'.
			if ( '=' === $la2 ) {
				$this->consume(); // Consume the '='.
				if ( '>' === $this->la( 3 ) ) {
					$this->consume(); // Consume the '>'.
					$this->type = self::NULL_SAFE_EQUAL_OPERATOR;
				} else {
					$this->type = self::LESS_OR_EQUAL_OPERATOR;
				}
			} elseif ( '>' === $la2 ) {
				$this->consume(); // Consume the '>'.
				$this->type = self::NOT_EQUAL_OPERATOR;
			} elseif ( '<' === $la2 ) {
				$this->consume(); // Consume the '<'.
				$this->type = self::SHIFT_LEFT_OPERATOR;
			} else {
				$this->type = self::LESS_THAN_OPERATOR;
			}
```